dashboard/src/actions/metadataTreeActions.js

```diff
@@ -0,0 +1,123 @@
+import * as TYPES from "./types";
+
+import { KEYS_JOIN_BY } from "assets/constants/overviewConstants";
+import store from "store/store";
+
+const { getState } = store;
+
+// Helper functions
+const isChecked = (dataItem, checkedItems) =>
+  checkedItems && checkedItems.some((item) => item === dataItem.key);
+const areAllDescendantsChecked = (dataItem, checkedItems) =>
+  dataItem.children
+    ? dataItem.children.every((child) => child.checkProps.checked === true)
+    : isChecked(dataItem, checkedItems);
+const areSomeDescendantsChecked = (dataItem, checkedItems) =>
+  dataItem.children
+    ? dataItem.children.some((child) =>
+        areSomeDescendantsChecked(child, checkedItems)
+      )
+    : isChecked(dataItem, checkedItems);
+const setChildNodes = (childNodes, isChecked) =>
+  childNodes.map((item) => {
+    item.checkProps.checked = isChecked;
+    return item;
+  });
+const getCheckedItemsKey = (item) => item.map((i) => i.key);
+const getIndexofKey = (arr, key) => arr.findIndex((item) => item.key === key);
+const updateChildKeysList = (checked, checkedItems, childKeys) => {
+  let keysList = [];
+  if (checked) {
+    keysList = [...checkedItems, ...childKeys];
+  } else {
+    keysList = checkedItems.filter((x) => !childKeys.includes(x));
+  }
+  return keysList;
+};
+
+export const mapTree = (item) => {
+  const checkedItems = getState().overview.checkedItems;
+  const hasCheck = areAllDescendantsChecked(item, checkedItems);
+
+  // Reset checked properties to be updated
+  item.checkProps.checked = false;
+
+  if (hasCheck) {
+    item.checkProps.checked = true;
+  } else {
+    const hasPartialCheck = areSomeDescendantsChecked(item, checkedItems);
+    if (hasPartialCheck) {
+      item.checkProps.checked = null;
+    }
+  }
+
+  if (item.children) {
+    return {
+      ...item,
+      children: item.children.map((child) => mapTree(child)),
+    };
+  }
+  return item;
+};
+
+export const onCheck =
+  (evt, treeViewItem, dataType) => async (dispatch, getState) => {
+    const checked = evt.target.checked;
+
+    const treeData = getState().overview.treeData;
+    let childKeys = [];
```

Comment:
`childKeys` doesn't seem to be referenced outside the block at line 75 where it is assigned; I recommend declaring it in that block and using `const` instead of `let`.

Reply:
Done